src/handle_action.cpp

```diff
@@ -1681,7 +1716,8 @@ bool game::handle_action()
             case ACTION_CLOSE:
                 if( u.has_active_mutation( trait_SHELL2 ) ) {
                     add_msg( m_info, _( "You can't close things while you're in your shell." ) );
-                } else if( u.has_effect( effect_riding ) ) {
+                } else if( u.has_effect( effect_riding ) && ( u.mounted_creature &&
+                           !u.mounted_creature.get()->has_flag( MF_RIDEABLE_MECH ) ) ) {
```

Comment:
I wonder why you used `u.mounted_creature.get()` directly here, but at many other places, you extract to a local variable `auto mon` first and use that one later.  Seems a bit inconsistent.